Mage/src/main/java/mage/game/events/CreateTokenEvent.java

```diff
@@ -17,11 +18,15 @@ public class CreateTokenEvent extends GameEvent {
      * @param source
      * @param controllerId
      * @param amount
-     * @param token
+     * @param tokensList
      */
-    public CreateTokenEvent(Ability source, UUID controllerId, int amount, Token token) {
+    public CreateTokenEvent(Ability source, UUID controllerId, int amount, List<Token> tokensList) {
         super(GameEvent.EventType.CREATE_TOKEN, null, source, controllerId, amount, false);
-        tokens.put(token, amount);
+        if (tokensList != null) {
```

Comment:
can tokensList be null? best if collections aren't null

Reply:
Currently, it's only initialized in TokenImpl where it can't be null without firing a Wrong Code Usage exception. But I figured defensive programming.